File: app/src/main/java/com/mojiscan/ocr/repository/TranscriptionRepository.java
package com.mojiscan.ocr.repository;

import android.app.Application;
import androidx.lifecycle.LiveData;
import com.mojiscan.ocr.data.dao.TranscriptionDao;
import com.mojiscan.ocr.data.database.AppDatabase;
import com.mojiscan.ocr.data.entity.TranscriptionEntity;

import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class TranscriptionRepository {
    private TranscriptionDao transcriptionDao;
    private ExecutorService executorService;

    public TranscriptionRepository(Application application) {
        AppDatabase database = AppDatabase.getDatabase(application);
        transcriptionDao = database.transcriptionDao();
        executorService = Executors.newSingleThreadExecutor();
    }

    public LiveData<List<TranscriptionEntity>> getAllTranscriptions() {
        return transcriptionDao.getAllTranscriptions();
    }

    public LiveData<List<TranscriptionEntity>> searchTranscriptions(String query) {
        return transcriptionDao.searchTranscriptions(query);
    }

    public TranscriptionEntity getTranscriptionById(long id) {
        return transcriptionDao.getTranscriptionById(id);
    }

    public void insertTranscription(TranscriptionEntity transcription) {
        executorService.execute(() -> transcriptionDao.insertTranscription(transcription));
    }

    public void updateTranscription(TranscriptionEntity transcription) {
        executorService.execute(() -> transcriptionDao.updateTranscription(transcription));
    }

    public void deleteTranscription(TranscriptionEntity transcription) {
        executorService.execute(() -> transcriptionDao.deleteTranscription(transcription));
    }

    public void deleteTranscriptionById(long id) {
        executorService.execute(() -> transcriptionDao.deleteTranscriptionById(id));
    }
}

File: app/src/main/java/com/mojiscan/ocr/repository/TranscriptionRepository.kt
package com.mojiscan.ocr.repository

import com.mojiscan.ocr.data.dao.TranscriptionDao
import com.mojiscan.ocr.data.entity.TranscriptionEntity
import kotlinx.coroutines.flow.Flow

class TranscriptionRepository(private val transcriptionDao: TranscriptionDao) {
    fun getAllTranscriptions(): Flow<List<TranscriptionEntity>> {
        return transcriptionDao.getAllTranscriptions()
    }

    fun searchTranscriptions(query: String): Flow<List<TranscriptionEntity>> {
        return transcriptionDao.searchTranscriptions(query)
    }

    suspend fun getTranscriptionById(id: Long): TranscriptionEntity? {
        return transcriptionDao.getTranscriptionById(id)
    }

    suspend fun insertTranscription(transcription: TranscriptionEntity): Long {
        return transcriptionDao.insertTranscription(transcription)
    }

    suspend fun updateTranscription(transcription: TranscriptionEntity) {
        transcriptionDao.updateTranscription(transcription)
    }

    suspend fun deleteTranscription(transcription: TranscriptionEntity) {
        transcriptionDao.deleteTranscription(transcription)
    }

    suspend fun deleteTranscriptionById(id: Long) {
        transcriptionDao.deleteTranscriptionById(id)
    }
}

